sigstore/sign.py

```diff
@@ -82,74 +89,106 @@ class Signer:
     The primary API for signing operations.
     """
 
-    def __init__(self, *, fulcio: FulcioClient, rekor: RekorClient):
+    def __init__(self, identity_token: str, cache: bool = True) -> None:
         """
         Create a new `Signer`.
 
-        `fulcio` is a `FulcioClient` capable of connecting to a Fulcio instance
-        and returning signing certificates.
+        `identity_token` is the identity token used to request a signing certificate
+        from Fulcio.
 
-        `rekor` is a `RekorClient` capable of connecting to a Rekor instance
-        and creating transparency log entries.
+        `cache` determines whether the signing certificate and ephemeral private key
+        should be reused (until the certificate expires) to sign different artifacts.
+        Default is `True`.
         """
-        self._fulcio = fulcio
-        self._rekor = rekor
+        self._identity_token: str = identity_token
+        self.cache: bool = cache
+        self._signing_certificate: Optional[FulcioCertificateSigningResponse] = None
+        self._private_key: Optional[ec.EllipticCurvePrivateKey] = None
+
+    @property
+    def private_key(self) -> ec.EllipticCurvePrivateKey:
+        """Get or generate a signing key."""
+        if not self.cache or self._private_key is None:
+            logger.debug("Generating ephemeral keys...")
+            self._private_key = ec.generate_private_key(ec.SECP384R1())
+
+        return self._private_key
```

Comment:
Small thing, but let's consider making the public API surface as small as possible to start with: `Signer` instances should only expose the `sign(...)` operation to begin with, not the helper properties that enable that operation.

Given that, this should probably be `def _private_key(self)` and the corresponding attribute should be `self.__private_key` (and similarly for `signing_cert`).